src/java_bytecode/ci_lazy_methods_needed.h

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************\
+
+Module: Context-insensitive lazy methods container
+
+Author: Chris Smowton, chris.smowton@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// Context-insensitive lazy methods container
+
+#ifndef CPROVER_JAVA_BYTECODE_CI_LAZY_METHODS_H
+#define CPROVER_JAVA_BYTECODE_CI_LAZY_METHODS_H
+
+#include <vector>
+#include <set>
+#include <util/symbol_table.h>
+
+class ci_lazy_methods_neededt
+{
+public:
+  ci_lazy_methods_neededt(
+    std::vector<irep_idt> &_needed_methods,
+    std::set<irep_idt> &_needed_classes,
+    symbol_tablet &_symbol_table):
+  needed_methods(_needed_methods),
+  needed_classes(_needed_classes),
+  symbol_table(_symbol_table)
+  {}
+
+  void add_needed_method(const irep_idt &);
```

Comment:
I think this is non-trivial and since not my code would prefer to leave. It would either trivially return true (since it always adds something to the vector) or I could swap it to a set which _feels_ right, but would change the behaviour.